lib/ap/string/mempcpy.c

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright (c) 2005-2014 Rich Felker, et al.
+ * Copyright (c) 2015-2016 Álvaro Jurado et al.
+ *
+ * Use of this source code is governed by a MIT-style
+ * license that can be found in the LICENSE.mit file.
+ */
+
+#define _GNU_SOURCE
+#include <string.h>
+
+void *mempcpy(void *dest, const void *src, size_t n)
+{
+	return (char *)memcpy(dest, src, n) + n;
```

Comment:
The implementation it's the same in BSD and musl. Glibc returns a void pointer. Reading Linux manual (it's a GNU implementation, non-POSIX):

`The mempcpy() function is nearly identical to the memcpy(3) function.
It copies n bytes from the object beginning at src into the object
pointed to by dest.  But instead of returning the value of dest it
returns a pointer to the byte following the last written byte.`

Seems that it's a kind of cast over the returning char.